internal/boxcli/midcobra/trace.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package midcobra
+
+import (
+	"context"
+	"os"
+	"runtime/trace"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type TraceMiddleware struct {
+	executionID string // uuid
+	tracef      *os.File
+	flag        *pflag.Flag
+	task        *trace.Task
+}
+
+var _ Middleware = (*DebugMiddleware)(nil)
+
+func (t *TraceMiddleware) AttachToFlag(flags *pflag.FlagSet, flagName string) {
+	flags.String(flagName, "", "write a trace to a file")
+	t.flag = flags.Lookup(flagName)
+	t.flag.Hidden = true
+	t.flag.NoOptDefVal = "trace.out"
+}
+
+func (t *TraceMiddleware) preRun(cmd *cobra.Command, args []string) {
+	if t == nil {
+		return
+	}
+	path := t.flag.Value.String()
+	if path == "" {
+		return
+	}
+	var err error
+	t.tracef, err = os.Create(path)
+	if err != nil {
+		panic("error enabling tracing: " + err.Error())
```

Comment:
should we panic here? why not log an error out but continue execution?

Reply:
I would've preferred to return an error, but I couldn't see an easy way of doing that with the middleware. I figured that if you're using the `--trace` flag you probably don't want things to continue if tracing can't be enabled.

I could alternatively log an error and then `os.Exit` if that seems cleaner?